pkg/http/api.go

```diff
@@ -1010,9 +1011,21 @@ func (a *api) onPublish(reqCtx *fasthttp.RequestCtx) {
 
 	err = a.publishFn(&req)
 	if err != nil {
+		status := fasthttp.StatusInternalServerError
 		msg := NewErrorResponse("ERR_PUBSUB_PUBLISH_MESSAGE",
 			fmt.Sprintf(messages.ErrPubsubPublishMessage, topic, pubsubName, err.Error()))
-		respondWithError(reqCtx, fasthttp.StatusInternalServerError, msg)
+
+		if errors.As(err, &runtime_pubsub.NotAllowedError{}) {
```

Comment:
I think we need to have the same change in grpc API, don't we?

Reply:
Yes of course. Standby.